src/CodeGeneration/OfferFriendlyOverloadsGenerator.cs

```diff
@@ -110,15 +110,23 @@ select new
                     .WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.None))
                     .WithExpressionBody(null);
 
-                var flags = GeneratorFlags.NativePointerToIntPtr;
-                var intPtrOverload = transformedMethodBase
-                    .WithParameterList(TransformParameterList(method.ParameterList, flags, parametersToFriendlyTransform))
-                    .WithBody(CallNativePointerOverload(method, flags, parametersToFriendlyTransform));
-                generatedType = generatedType.AddMembers(intPtrOverload);
+                var preexistingMethod = from newMethod in generatedType.Members.OfType<MethodDeclarationSyntax>()
+                                        where WhereIsIntPtrParameter(method.ParameterList.Parameters, newMethod.ParameterList.Parameters).Any() && newMethod.Identifier.ValueText == transformedMethodBase.Identifier.ValueText
+                                        select method;
+
+                if (!preexistingMethod.Any())
+                {
+                    var flags = GeneratorFlags.NativePointerToIntPtr;
+                    var intPtrOverload = transformedMethodBase
+                        .WithParameterList(TransformParameterList(method.ParameterList, flags, parametersToFriendlyTransform))
+                        .WithBody(CallNativePointerOverload(method, flags, parametersToFriendlyTransform));
+                    generatedType = generatedType.AddMembers(intPtrOverload);
```

Comment:
That is from Roslyn's API [`ClassDeclarationSyntax.AddMembers`](http://www.nudoq.org#!/Packages/Roslyn.Compilers.CSharp/Roslyn.Compilers.CSharp/ClassDeclarationSyntax/M/AddMembers). Are you suggesting we create class extensions or a path on Roslyn classes?